main/src/cgeo/geocaching/filters/core/StatusGeocacheFilter.java

```diff
@@ -39,18 +56,40 @@
     private boolean excludeDisabled = false;
     private boolean excludeArchived = false;
 
-    private Boolean statusOwn = null;
+
+    private Boolean statusOwned = null;
     private Boolean statusFound = null;
+    private Boolean statusStored = null;
+    private Boolean statusFavorite = null;
+    private Boolean statusWatchlist = null;
+    private Boolean statusPremium = null;
+    private Boolean statusSolvedMystery = null;
 
 
     @Override
     public Boolean filter(final Geocache cache) {
+        //handle a few "inconclusive" cases
+        if ((statusFavorite != null && cache.isFavoriteRaw() == null) ||
+            (statusWatchlist != null && cache.isOnWatchlistRaw() == null) ||
+            (statusPremium != null && cache.isPremiumMembersOnlyRaw() == null) ||
+            (statusSolvedMystery != null && cache.getUserModifiedCoordsRaw() == null)) {
+            return null;
+        }
+
         return
             ((!excludeActive && !cache.isDisabled() && !cache.isArchived()) ||
                 (!excludeDisabled && cache.isDisabled()) ||
                 (!excludeArchived && cache.isArchived())) &&
-            (statusOwn == null || (cache.isOwner() == statusOwn)) &&
-            (statusFound == null || cache.isFound() == statusFound);
+            (statusOwned == null || (cache.isOwner() == statusOwned)) &&
+            (statusFound == null || cache.isFound() == statusFound) &&
+            (statusStored == null || cache.isOffline() == statusStored) &&
+            (statusFavorite == null || cache.isFavorite() == statusFavorite) &&
+            (statusWatchlist == null || cache.isOnWatchlist() == statusWatchlist) &&
+            (statusPremium == null || cache.isPremiumMembersOnly() == statusPremium) &&
+            (statusSolvedMystery == null || cache.getType() != CacheType.MYSTERY ||
+                (cache.hasUserModifiedCoords() ||
```

Comment:
@eddiemuc there is already a function in Geocache.java to check that. See GeoCache.showUserModifiedCoords. That should give you the desired information?

Reply:
I am very sorry, I currently don't understand  your remark.

In above code, `cache` is a `Geocache` object, and I use its method `hasUserModifiedCoords()` for my check. I couldn't find a method `showUserModifiedCoords()` though. 

Method `hasUserModifiedCoords()` does not check for valid final coords, I use `Geocache.getFirstMatchingWaypoint()` for that (which I created as part of this PR).

How do you think this should be done instead?

Reply:
Ah ok, my mistake, thought it was a function of the class Geocache, but it is a private one in MapMarkerUtils: showUserModifiedCoords

There is a method in Geocache.hasFinalDefined, then you don't have to lookup that ...

Reply:
Ah, you're right, did not see this one. Good catch, thank you. I will change this in one of my next PRs (since this one is already merged)